internal/endpoints/builtins.go

```diff
@@ -0,0 +1,145 @@
+package endpoints
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+type BuiltInParameterResolver interface {
+	ResolveBuiltIn(name string) (value interface{}, ok bool)
+}
+
+type BuiltInResolver struct {
+	Region string
+
+	UseFIPS aws.FIPSEndpointState
+
+	UseDualStack aws.DualStackEndpointState
+
+	ForcePathStyle bool
+
+	Accelerate bool
+
+	DisableMultiRegionAccessPoints bool
+
+	Endpoint *url.URL
+
+	S3UseArnRegion bool
+
+	S3ControlUseArnRegion bool
+}
+
+type NopBuiltInResolver struct{}
+
+func (b *NopBuiltInResolver) ResolveBuiltIn(name string) (value interface{}, ok bool) {
+	return nil, true
+}
+
+func (b *BuiltInResolver) ResolveBuiltIn(name string) (value interface{}, ok bool) {
```

Comment:
Why do we need this separate structure with string-based input for builtins?

We know where everything should be at codegen time, we should be able to just generate a static builtin resolver method per-service (per-service because each `Options` struct is per-service)

i.e.
```
func (e *EndpointParameters) bindBuiltins(o *Options) {
    // codegen each binding
}